### DISCUSSION TITLE
Create folder structure for tags to write based on messages coming from other application.

### DISCUSSION THREAD

**Discussion by jigar88** [FreeOpcUa/python-opcua#1262] [General]
Hello all,

I am working on creating opc server where I am receiving messages from other server. I am receiving continuous messages like "tagPath=AB/ABC1/Machine1/MachineNumber/Folder/Stopping|value=true|timestamp=2006-01-01T11:11:11Z07:00"
"tagPath=BC/BCD1/Machine2/MachineNumber/Folder/Stopped|value=false|timestamp=2006-01-01T11:11:13Z07:00"
"tagPath=BC/BCD1/Machine3/MachineNumber/Folder/Starting|value=true|timestamp=2006-01-01T11:11:15Z07:00" .

I am trying to creating structure like this with each message parsing but its not working properly. 

```
AB (object)
 ABC1 (object)
  Machine1 (object)
   MachineNumber (object)
    Folder (object)
	 Stopping (tag)
	 
BC (object)
 BCD1 (object)
  Machine2 (object)
   MachineNumber (object)
    Folder (object)
	 Stopped (tag)
  Machine3  (object)
   MachineNumber  (object)
    Folder  (object)
	 Starting (tag)
```

so for example in tagPath I am split the string with "/" and get the members and based on that list I want to create object folders hierarchy, lot of time it get repeats only tag value get changes so in that case I am just updating tag values. Is there an example I can look up , any help will be appreciated. Thanks.

**Comment by AndreasHeine** [comment]
dynamic addressspaces are tricky... setup nodes dynamically makes client application komplex and inefficient lots of "browsing" and "try reads". there is off course no example because its an unusual usecase for opc ua but it sounds possible to do!

we just provide the opc ua stack so the basic tools to build opcua-applications. what you try to do needs propper planing! you need to think about a lot of things how to create the nodetree for the message, how to update an existing nodetree, how to avoid memory leaks (if there is no message/nodetree livetime). you should start break the problem down till you can solve a problem and develope incrementally step by step! if you want to make a folderstructure you should use the folder-object-type instead of the base-object-type

python-opcua maintained by volunteers so actually no one will write that for you we can only help if you have specific questions or opcua related errors

**Reply by jigar88** [reply]
Thanks for guidance.  I will follow this approach. one last question is there a way to check if object created is already there, for instance if my ABC1 folder is already there?

**Reply by AndreasHeine** [reply]
you need to read more about opc ua and its services and concepts

https://reference.opcfoundation.org/v104/
https://reference.opcfoundation.org/v104/Core/docs/Part4/
